### PR TITLE
feat: implement e2e test for CompanyResolver

### LINKED ISSUE #829: Add integration tests on server
We don't have tests on server right now which will lead to regressions. 
As our first testing strategy, we would like to put in place integration tests: create test suite that will query the API and use a seeded postgres database

Todo:
- find a way to provision a testing postgres database locally (maybe just create a test database on the existing instance?)
- find a way to provision a testing prostgres database in CI
- write a script to seed the test database before running the tests
- write a first test suite. Make sure it's idempotent

### PR DESCRIPTION
**Pull request by magrinj**
This PR implements an end-to-end test for all methods in the `CompanyResolver`. The challenge testing this method was that it involves `AuthWorkspace`, a custom decorator, that extracts the authenticated workspace from the request.
So `user` and `workspace` are mocked for testing purpose.

### CODE REVIEW

File: server/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# docker-compose.yml
```
Comment (charlesBochet) [inline server/docker-compose.yml:1]: If possible we should avoid forcing users to use docker locally :
- developers should already have a postgres database up, maybe we should create a "test" schema during initialization (like we create the default schema)
- for ci, we should move this to infra/ci

File: server/package.json [hunk 2]
```diff
@@ -18,7 +18,7 @@
     "test:watch": "jest --watch",
     "test:cov": "jest --coverage",
     "test:debug": "node --inspect-brk -r tsconfig-paths/register -r ts-node/register node_modules/.bin/jest --runInBand",
-    "test:e2e": "jest --config ./test/jest-e2e.json",
+    "test:e2e": "./scripts/run-integration.sh",
```
Comment (charlesBochet) [inline server/package.json:21]: I think this run-integration using docker-compose should only be for CI
for devs, we should run against a pre-provisionned "test" database. We only need to use the right .env and the server should directly connect to it

**Review by charlesBochet — COMMENTED** [review]
Looks promising!
Added a comment:
- we should use a "test" database locally, no need to provision an extra postgres db through docker. We can use the same postgres as "default" database
- for CI, it's useful, let's move the docker-compose to infra/ci

Let's also maybe introduce two commands in package.json, one for local (just running test), one for CI (doing the postgres provisionning). Also, it might exists a simpler way to host a postgres database in .github actions. You might not need all this scripting: https://docs.github.com/en/actions/using-containerized-services/creating-postgresql-service-containers

**Review by charlesBochet — APPROVED** [review]
Big piece, thanks!

File: .github/workflows/ci-server.yaml [hunk 3]
```diff
@@ -16,6 +16,7 @@ jobs:
         env:
           POSTGRES_HOST: postgres
           POSTGRES_PASSWORD: postgrespassword
+          POSTGRES_DB: test
```
Comment (magrinj) [inline .github/workflows/ci-server.yaml:19]: @charlesBochet Database name was set tu plural in env: `tests`

**Pull request merged**